### ISSUE TITLE
Support try/except in partially defined check

### ISSUE THREAD

**Issue by ilinum** [python/mypy#13928]
Currently, partially defined check (`--enable-error-code partially-defined`) does not support detecting partially defined variables try/except statement. 

The most basic example is the following:
```python
try:
  x = foo()
except:
  pass
y = x  # mypy should generate an error here but does not.
```

**Comment by ilevkivskyi** [comment]
@ilinum What is the situation here? I see there were no activity on the PR last week. We will probably start 1.0 release process in a week or so. I would like to announce this feature in the release, and it would be great if `try/except` is properly supported.

**Comment by JukkaL** [comment]
I can do another round of review and make some performance measurements.
